lib/charms/prometheus_k8s/v1/prometheus.py

```diff
@@ -317,99 +314,71 @@ def __init__(self, charm, name, consumes, multi=False):
                 dictionary are corresponding minimal acceptable
                 semantic version specfications for the monitoring
                 service.
+            service: string name of Pebble service of consumer charm.
+            config: a optional dictionary with keys that are
+                Prometheus scrape configuration options, for example
+                metrics path and port.
             multi: an optional (default False) flag to indicate if
                 this object must support interaction with multiple
                 Prometheus monitoring service providers.
-
         """
         super().__init__(charm, name, consumes, multi)
+
         self._charm = charm
+        self._service = service
         self._relation_name = name
-        self._stored.set_default(targets={})
+        self._static_scrape_port = config.get("static_scrape_port", 80)
+        self._static_scrape_path = config.get("static_scrape_path", "/metrics")
+        self._multi_mode = multi
+
         events = self._charm.on[self._relation_name]
-        self.framework.observe(events.relation_joined, self._set_targets)
+        self.framework.observe(events.relation_joined, self._set_scrape_metadata)
```

Comment:
I'm pretty sure this could be done on `events.relation_created` which would get the info over to the other side a little bit sooner.  Not a big deal, though.

Reply:
Thanks. Interesting. Had not thought of that. Will investigate. And perhaps add it into next PR or this one.

Reply:
On further thought I actually need the relation joined because I am setting some unit data also. Leaving as is for now, but will keep this in mind for future.